pkg/service/schema.go

```diff
@@ -52,20 +54,50 @@ type SimpleParameterSchema struct {
 // ObjectParameterSchema represents the attributes of a complicated schema type
 // that can have nested properties
 type ObjectParameterSchema struct {
-	Type               string                     `json:"type"`
-	Description        string                     `json:"description,omitempty"`
-	Required           bool                       `json:"-"`
-	RequiredProperties []string                   `json:"required,omitempty"`
-	Properties         map[string]ParameterSchema `json:"properties,omitempty"`
-	Additional         ParameterSchema            `json:"additionalProperties,omitempty"` // nolint: lll
+	Description        string
+	Required           bool
+	requiredProperties []string
+	Properties         map[string]ParameterSchema
+	Additional         ParameterSchema
+}
+
+// MarshalJSON provides functionality to marshal an
+// ObjectParameterSchema to JSON
+func (o *ObjectParameterSchema) MarshalJSON() ([]byte, error) {
+	return json.Marshal(struct {
+		Type               string                     `json:"type"`
+		Description        string                     `json:"description,omitempty"`
+		RequiredProperties []string                   `json:"required,omitempty"`
+		Properties         map[string]ParameterSchema `json:"properties,omitempty"`
+		Additional         ParameterSchema            `json:"additionalProperties,omitempty"` // nolint: lll
+	}{
+		Type:               "object",
+		Description:        o.Description,
+		RequiredProperties: o.requiredProperties,
```

Comment:
I'm wondering if this is a better place than `NewService(...)` to walk the schemas for the constituent properties to build the `requiredProperties` array.

Any thoughts? I don't feel strongly one way or the other. It's just something that occurred to me.

Reply:
Ahh yeah. I meant to move that _back_ into the `addProperties` method and had forgotten! I've moved that back in there.  It was orignally in the `addProperties` on the interface and when i nuked it I forgot to just make the `addProperties(..)` recursively handle the new things being added to the top level schema.